src/test/java/org/codeforamerica/messaging/providers/mailgun/MailgunCallbackControllerTest.java

```diff
@@ -197,12 +200,6 @@ public void whenUnsubscribed_ThenSavesEmailSubscription() throws Exception {
                                     }
                                 """.formatted(recipient, TestData.PROVIDER_MESSAGE_ID)))
                 .andExpect(MockMvcResultMatchers.status().isOk());
-        ArgumentCaptor<EmailSubscription> emailSubscriptionCaptor = ArgumentCaptor.forClass(EmailSubscription.class);
-        Mockito.verify(emailSubscriptionRepository).save(emailSubscriptionCaptor.capture());
-        EmailSubscription emailSubscription = emailSubscriptionCaptor.getValue();
-        assertTrue(emailSubscription.isUnsubscribed());
-        assertEquals(emailSubscription.getEmail(), recipient);
-        assertTrue(emailSubscription.isSourceInternal());
-        assertNotEquals(MessageStatus.unsubscribed, message.getEmailStatus());
+        Mockito.verify(emailService).unsubscribe(recipient);
```

Comment:
Would it make sense to also check the return value of `emailService.unsubscribed` here? Or is that covered in another test?

Reply:
No, because the service is mocked. Yes, it is covered in `org/codeforamerica/messaging/services/EmailServiceTest.java`.